src/build123d/topology.py

```diff
@@ -2125,7 +2494,7 @@ def combined_center(
                 sum_wc = sum_wc.add(weighted_center)
             middle = Vector(sum_wc.multiply(1.0 / total_mass))
         elif center_of == CenterOf.BOUNDING_BOX:
-            total_mass = len(objects)
+            total_mass = len(list(objects))
 
             weighted_centers = []
             for obj in objects:
```

Comment:
if the iterable happens to be lazy `len(list(objects))` would consume it and `for obj in objects:` would have nothing to loop on
```suggestion
            total_mass = 0
            weighted_centers = []
            for obj in objects:
                total_mass += 1
```

Reply:
I've checked them all and in a few cases converted the inputs to lists explicitly. Thanks for the pointer.